src/Features/Core/Diagnostics/WorkspaceAnalyzerManager_HostDiagnostics.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics.Log;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal sealed partial class WorkspaceAnalyzerManager
```

Comment:
I am going to just move these methods directly to AbstractHostDiagnosticUpdateSource. I just parked it onto WorkspaceAnalyzerManager thinking it could be central place for analyzer related things to look for, but you are right it probably doesn't lie here.